### PR TITLE
285385: Fixes for software templates

### PR DESCRIPTION
**Pull request by jeevankuduvaravindran**
# **What this PR does / why we need it**:
This PR contains the below fixes for software templates.

- UI scaffolder working now. Fixed the templating syntax issues with nunjucks and backstage.
- Fixes to frontend home nunjucks template 
- Templating issues for deployFromFeature in build.yaml

[AB#285385](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/285385)

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
Tested the scaffolder locally to scaffold a frontend repo from backstage.

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: templates/adp-backend-template-node/skeleton/.azuredevops/build.yaml
Comment (kbabigumira) [inline templates/adp-backend-template-node/skeleton/.azuredevops/build.yaml:1]: Variable groups should be made optional.  @rajeshkaremane  @jeevankuduvaravindran   suggest we remove them from the template but add documentation that explains how to use variable groups.

As is, pipeline does not run successfully if the variable groups have not been created.  Not every service will have external secrets.

File: templates/adp-frontend-template-node/skeleton/.azuredevops/build.yaml
Comment (rajeshkaremane) [inline templates/adp-frontend-template-node/skeleton/.azuredevops/build.yaml:1]: Better to comment the variableGroups and Variables secttion, so it is available just incase they want to use it

File: templates/adp-backend-template-node/skeleton/.azuredevops/build.yaml
Comment (jeevankuduvaravindran) [reply templates/adp-backend-template-node/skeleton/.azuredevops/build.yaml:1]: @kbabigumira updated the build.yaml and readme accordingly.

File: templates/adp-frontend-template-node/skeleton/.azuredevops/build.yaml
Comment (jeevankuduvaravindran) [reply templates/adp-frontend-template-node/skeleton/.azuredevops/build.yaml:1]: @rajeshkaremane done

**Review by rajeshkaremane — APPROVED** [review]

**Pull request merged**